packages/cli/src/environments/variables/variables.service.ee.ts

```diff
@@ -32,14 +34,32 @@ export class EEVariablesService extends VariablesService {
 		this.validateVariable(variable);
 
 		void Container.get(InternalHooks).onVariableCreated({ variable_type: variable.type });
-		return collections.Variables.save(variable);
+		return collections.Variables.save({
+			...variable,
+			id: generateNanoId(),
+		});
 	}
 
-	static async update(id: number, variable: Omit<Variables, 'id'>): Promise<Variables> {
+	static async update(id: string, variable: Omit<Variables, 'id'>): Promise<Variables> {
 		this.validateVariable(variable);
 
-		await collections.Variables.update(id, variable);
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		return (await this.get(id))!;
+		const existingVariable = await this.get(id);
+		if (!existingVariable) {
+			throw new Error('Variable not found');
+		} else {
+			if (existingVariable.key !== variable.key) {
+				const newVariable = new Variables(variable);
+				await Db.transaction(async (transactionManager) => {
+					await transactionManager.save(Variables, newVariable);
+					await transactionManager.delete(Variables, id);
+				});
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				return (await this.get(newVariable.id))!;
+			} else {
+				await collections.Variables.update(id, variable);
+			}
```

Comment:
Why can't we simply perform an update in all cases?

Reply:
Because we have this strange case where we have unique constraints on both key and id, but seperate from another. 
This resulted in potential conflicts when a key is renamed or imported from somewhere else with the same key as an existing variable. I can check again whether there's another way to avoid this, but i'm pretty sure I ran into a few edge cases where the unique constraints were fighting each other in a way that only complete reconstruction would do. Will have to double check.

Reply:
Gotcha, makes sense. Thanks for clarifying